core/src/impl/Kokkos_Core.cpp

```diff
@@ -70,7 +71,12 @@ bool is_unsigned_int(const char* str)
   }
   return true;
 }
-
+void convert_uppercase(char* str)
+{
+  const size_t len = strlen (str);
```

Comment:
See note below.  (Modifying result of `std::getenv` is undefined behavior, i.e., don't do it in code that aims to be portable.)